tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/cocosnet_evaluator.py

```diff
@@ -0,0 +1,522 @@
+                    predictions,
+                    metrics_result=metrics_result,
+                    element_identifiers=batch_identifiers,
+                    dataset_indices=batch_input_ids
+                )
+
+            if self.metric_executor.need_store_predictions:
+                self._annotations.extend(gan_annotations)
+                self._predictions.extend(gan_predictions)
+
+            if _progress_reporter:
+                _progress_reporter.update(batch_id, len(batch_predictions))
+                if compute_intermediate_metric_res and _progress_reporter.current % metric_interval == 0:
+                    self.compute_metrics(
+                        print_results=True, ignore_results_formatting=ignore_results_formatting
+                    )
+
+        if _progress_reporter:
+            _progress_reporter.finish()
+
+        return self._annotations, self._predictions
+
+    def compute_metrics(self, print_results=True, ignore_results_formatting=False):
+        if self._metrics_results:
+            del self._metrics_results
+            self._metrics_results = []
+
+        for result_presenter, evaluated_metric in self.metric_executor.iterate_metrics(
+                self._annotations, self._predictions):
+            self._metrics_results.append(evaluated_metric)
+            if print_results:
+                result_presenter.write_result(evaluated_metric, ignore_results_formatting)
+        return self._metrics_results
+
+    def extract_metrics_results(self, print_results=True, ignore_results_formatting=False):
+        if not self._metrics_results:
+            self.compute_metrics(False, ignore_results_formatting)
+
+        result_presenters = self.metric_executor.get_metric_presenters()
+        extracted_results, extracted_meta = [], []
+        for presenter, metric_result in zip(result_presenters, self._metrics_results):
+            result, metadata = presenter.extract_result(metric_result)
+            if isinstance(result, list):
+                extracted_results.extend(result)
+                extracted_meta.extend(metadata)
+            else:
+                extracted_results.append(result)
+                extracted_meta.append(metadata)
+            if print_results:
+                presenter.write_result(metric_result, ignore_results_formatting)
+
+        return extracted_results, extracted_meta
+
+    def print_metrics_results(self, ignore_results_formatting=False):
+        if not self._metrics_results:
+            self.compute_metrics(True, ignore_results_formatting)
+            return
+        result_presenters = self.metric_executor.get_metric_presenters()
+        for presenter, metric_result in zip(result_presenters, self._metrics_results):
+            presenter.write_result(metric_result, ignore_results_formatting)
+
+    def reset_progress(self, progress_reporter):
+        progress_reporter.reset(self.dataset.size)
+
+    def release(self):
+        self.launcher.release()
```

Comment:
models also should be released

Reply:
Now all models are released. But when I add release for `check_model` and `test_model.generator`, linux raise `Segmentation fault (core dumped)`. The error occurs due to a release, but I don't see any mistakes in the code. Similar construction in colorization network works without errors, but there only 2 models are released (in cocosnet 3). Maybe it is a bug on openvino inference engine.

Reply:
which order do you use?
Please note, it is important to remove models before launcher.
Also check that you do not remove the same entity twice (e.g. IECore)